src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -93,4 +99,139 @@ public static String toQueryString(Map<String, String> params) {
         result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
         return result;
     }
+
+    /**
+     * TODO: fully implement
+     */
+    public static boolean itineraryIsSavable(Itinerary itinerary) {
+        return true;
+    }
+
+    /**
+     * Returns true if the itineraries match for the purposes of trip monitoring.
+     *
+     * @param previousItinerary The original itinerary that others are compared against.
+     * @param newItinerary A new itinerary that might match the previous itinerary.
+     */
+    public static boolean itinerariesMatch(Itinerary previousItinerary, Itinerary newItinerary) {
+        // make sure either itinerary is monitorable before continuing
+        if (!itineraryIsSavable(previousItinerary) || !itineraryIsSavable(newItinerary)) return false;
+
+        // make sure itineraries have same amount of legs
+        if (previousItinerary.legs.size() != newItinerary.legs.size()) return false;
```

Comment:
For this return value and the `legsMatch` return value, it might be nice to have a reason returned. This is non-blocking and can probably happen in a later PR.

Reply:
ok, will do in another PR